src/main/webapp/js/utils/table-cells/text-cell.jsx

```diff
@@ -1,9 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const TextCell = ({ value }) => (
+const TextCell = ({ value, format, pattern }) => (
   <div className="table-cell-text">
-    {value}
+    {!format || !pattern || !value ? value : value.replace(/\D+/g, '').replace(new RegExp(pattern), format)}
```

Comment:
I think it will be best to put the first part in `()`, also remove the first replace (this will not be only used for phone numbers). I think you can add the `i` flag by default to the RegExp constructor.